Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/preprocessor/CqlPreprocessorElmCommonVisitor.kt

```diff
@@ -0,0 +1,878 @@
+@file:Suppress("WildcardImport")
+
+package org.cqframework.cql.cql2elm.preprocessor
+
+import jakarta.xml.bind.JAXBElement
+import java.io.Serializable
+import java.util.*
+import javax.xml.namespace.QName
+import org.antlr.v4.runtime.ParserRuleContext
+import org.antlr.v4.runtime.TokenStream
+import org.antlr.v4.runtime.misc.Interval
+import org.antlr.v4.runtime.tree.ParseTree
+import org.antlr.v4.runtime.tree.TerminalNode
+import org.apache.commons.lang3.tuple.Pair
+import org.cqframework.cql.cql2elm.*
+import org.cqframework.cql.cql2elm.model.Chunk
+import org.cqframework.cql.cql2elm.model.FunctionHeader
+import org.cqframework.cql.cql2elm.model.Model
+import org.cqframework.cql.elm.IdObjectFactory
+import org.cqframework.cql.elm.tracking.TrackBack
+import org.cqframework.cql.elm.tracking.Trackable
+import org.cqframework.cql.gen.cqlBaseVisitor
+import org.cqframework.cql.gen.cqlParser.*
+import org.hl7.cql.model.*
+import org.hl7.cql_annotations.r1.Annotation
+import org.hl7.cql_annotations.r1.Narrative
+import org.hl7.cql_annotations.r1.ObjectFactory
+import org.hl7.cql_annotations.r1.Tag
+import org.hl7.elm.r1.*
+
+/** Common functionality used by [CqlPreprocessor] and [Cql2ElmVisitor] */
+@Suppress(
+    "LargeClass",
+    "CyclomaticComplexMethod",
+    "NestedBlockDepth",
+    "TooManyFunctions",
+    "ComplexCondition",
+    "ImplicitDefaultLocale",
+    "ReturnCount"
+)
+open class CqlPreprocessorElmCommonVisitor(
+    libraryBuilder: LibraryBuilder,
+    tokenStream: TokenStream
+) : cqlBaseVisitor<Any?>() {
+    @JvmField protected val of: IdObjectFactory
+    protected val af = ObjectFactory()
+    protected var implicitContextCreated = false
+    @JvmField protected var currentContext = "Unfiltered"
+    @JvmField protected var chunks = Stack<Chunk>()
+    @JvmField protected val libraryBuilder: LibraryBuilder
+    protected val tokenStream: TokenStream
+    var libraryInfo = LibraryInfo()
+        protected set
+
+    var isAnnotationEnabled = false
+        private set
+
+    var isDetailedErrorsEnabled = false
+        private set
+
+    private var locate = false
+    private var resultTypes = false
+    var dateRangeOptimization = false
+        private set
+
+    var isMethodInvocationEnabled = true
+        private set
+
+    var isFromKeywordRequired = false
+        private set
+
+    @JvmField var includeDeprecatedElements = false
+
+    init {
+        this.libraryBuilder = Objects.requireNonNull(libraryBuilder, "libraryBuilder required")
```

Comment:
this initialization is no longer needed, since Kotlin is non-nullable by default

Reply:
:+1: I've removed this line because `libraryBuilder` is declared as `LibraryBuilder`